gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionExtension.java

```diff
@@ -89,15 +90,23 @@ public JavaServiceDistributionExtension(Project project) {
         return mainClass;
     }
 
-    public final void setMainClass(String mainClass) {
+    public final void mainClass(String newMainClass) {
+        this.mainClass.set(newMainClass);
+    }
+
+    public final void setMainClass(Provider<String> mainClass) {
```

Comment:
I don't think we should expose such a method. We should follow the pattern that other gradle task/extensions have set and only expose a getter to the underlying property. ex: https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/bundling/AbstractArchiveTask.html